### PR TITLE
RFQv2 use v5 sig checker

### PR DESCRIPTION
**Pull request by charlesjhongc**
In order to support old MMP wallet, still accept multiple wallet signature validation type in RFQv2. Also, add more tests.

### CODE REVIEW

**Comment by charlesjhongc** [comment]
Depends on #187

File: test/forkMainnet/RFQv2.t.sol [hunk 1]
```diff
@@ -68,10 +78,10 @@ contract RFQTest is StrategySharedSetup {
             salt: defaultSalt
         });
 
-        defaultMakerSig = _signOffer(makerPrivateKey, defaultOffer);
+        defaultMakerSig = _signOffer(makerPrivateKey, defaultOffer, SignatureValidator.SignatureType.EIP712);
 
         defaultOrder = RFQOrder({ offer: defaultOffer, recipient: payable(recipient), feeFactor: defaultFeeFactor });
-        defaulttakerSig = _signRFQOrder(takerPrivateKey, defaultOrder);
+        defaulttakerSig = _signRFQOrder(takerPrivateKey, defaultOrder, SignatureValidator.SignatureType.EIP712);
```
Comment (pilagod) [inline test/forkMainnet/RFQv2.t.sol:84]:
```suggestion
        defaultTakerSig = _signRFQOrder(takerPrivateKey, defaultOrder, SignatureValidator.SignatureType.EIP712);
```

File: test/forkMainnet/RFQv2.t.sol [hunk 2]
```diff
@@ -315,26 +437,58 @@ contract RFQTest is StrategySharedSetup {
...
+            sigType == SignatureValidator.SignatureType.WalletBytes32
+        ) {
+            (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, EIP712SignDigest);
+            sig = abi.encodePacked(r, s, v, uint8(sigType));
+        } else if (sigType == SignatureValidator.SignatureType.Wallet) {
+            (uint8 v, bytes32 r, bytes32 s) = vm.sign(_privateKey, ECDSA.toEthSignedMessageHash(EIP712SignDigest));
+            sig = abi.encodePacked(r, s, v, uint8(sigType));
+        } else {
+            revert("Invalid signature type");
+        }
```
Comment (pilagod) [inline test/forkMainnet/RFQv2.t.sol:491]: nit: This part is kind of duplicate to those in `_signOffer`. Maybe we could extract a common `_signDigest(uint256, bytes, SignatureType)` to get rid of the duplicate.

File: test/forkMainnet/RFQv2.t.sol (hunk 1 above)
Comment (charlesjhongc) [reply test/forkMainnet/RFQv2.t.sol]: Fixed in 9de0aba

File: test/forkMainnet/RFQv2.t.sol (hunk 2 above)
Comment (charlesjhongc) [reply test/forkMainnet/RFQv2.t.sol]: Sounds good. Fixed in cf380bc.

**Review by pilagod — APPROVED** [review]

**Review by NIC619 — APPROVED** [review]

**Pull request merged**